Overloading of arithmatic operator in type-alias

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**

```
V full version: V 0.4.4 4bc9a8f.28e810d
OS: macos, macOS, 14.0, 23A344
Processor: 8 cpus, 64bit, little endian, Apple M2

getwd: /Users/andreas/dev/vlangtest/typealiasoverload
vexe: /Users/andreas/v/v
vexe mtime: 2024-01-10 13:08:26

vroot: OK, value: /Users/andreas/v
VMODULES: OK, value: /Users/andreas/.vmodules
VTMP: OK, value: /tmp/v_501

Git version: git version 2.39.3 (Apple Git-145)
Git vroot status: 0.4.4-3-g28e810d9
.git/config present: true

CC version: Apple clang version 15.0.0 (clang-1500.0.40.1)
thirdparty/tcc status: thirdparty-macos-arm64 a668e5a0

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg main.v`
```v
type Vec3 = [3]int

fn (v Vec3) add (u Vec3) Vec3 {
	return Vec3([v[0] + u[0], v[1] + u[1], v[2] + u[2]]!)
}

fn (v Vec3) + (u Vec3) Vec3 {
	return Vec3([v[0] + u[0], v[1] + u[1], v[2] + u[2]]!)
}

fn main() {
	vec := Vec3([1, 2, 3]!)
	dump(vec.add(vec))
	dump(vec + vec)
	println('Hello world!')
}
```

**What did you expect to see?**

Overloading of + operator results in an error: 'struct _v_Array_fixed_int_3' to 'int *'
Add method works just fine.

Overloading arithmatic operator on type-alias

**What did you see instead?**
```
==================
              ^
/tmp/v_501/main.1832658909320647278.tmp.c:12744:68: error: passing '_v_Array_fixed_int_3' (aka 'struct _v_Array_fixed_int_3') to parameter of incompatible type 'int *'
         _v_dump_expr_main__Vec3(_SLIT("main.v"), 14, _SLIT("vec + vec"), main__Vec3__plus(vec, vec));
                                                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/v_501/main.1832658909320647278.tmp.c:2124:102: note: passing argument to parameter 'dump_arg' here
_v_Array_fixed_int_3 _v_dump_expr_main__Vec3(string fpath, int line, string sexpr, Array_fixed_int_3 dump_arg) {
                                                                                                     ^
2 warnings and 1 error generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/assign.v
+++ b/vlib/v/gen/c/assign.v
@@ -369,7 +369,7 @@ fn (mut g Gen) assign_stmt(node_ ast.AssignStmt) {
 		right_sym := g.table.sym(unwrapped_val_type)
 		unaliased_right_sym := g.table.final_sym(unwrapped_val_type)
 		is_fixed_array_var := unaliased_right_sym.kind == .array_fixed && val !is ast.ArrayInit
-			&& (val in [ast.Ident, ast.IndexExpr, ast.CallExpr, ast.SelectorExpr, ast.DumpExpr]
+			&& (val in [ast.Ident, ast.IndexExpr, ast.CallExpr, ast.SelectorExpr, ast.DumpExpr, ast.InfixExpr]
 			|| (val is ast.CastExpr && val.expr !is ast.ArrayInit)
 			|| (val is ast.PrefixExpr && val.op == .arrow)
 			|| (val is ast.UnsafeExpr && val.expr is ast.Ident)) && !g.pref.translated

--- a/vlib/v/gen/c/infix.v
+++ b/vlib/v/gen/c/infix.v
@@ -807,6 +807,12 @@ fn (mut g Gen) infix_expr_arithmetic_op(node ast.InfixExpr) {
 			g.op_arg(node.right, method.params[1].typ, right.typ)
 		}
 		g.write(')')
+
+		if left.typ != 0 && !left.typ.has_flag(.option) && !left.typ.has_flag(.result)
+			&& g.table.final_sym(left.typ).kind == .array_fixed {
+			// it's non-option fixed array, requires accessing .ret_arr member to get the array
+			g.write('.ret_arr')
+		}
 	}
 }
 

--- a/vlib/v/tests/fixed_array_op_overload_test.v
+++ b/vlib/v/tests/fixed_array_op_overload_test.v
@@ -1,0 +1,16 @@
+type Vec3 = [3]int
+
+fn (v Vec3) add(u Vec3) Vec3 {
+	return Vec3([v[0] + u[0], v[1] + u[1], v[2] + u[2]]!)
+}
+
+fn (v Vec3) + (u Vec3) Vec3 {
+	return Vec3([v[0] + u[0], v[1] + u[1], v[2] + u[2]]!)
+}
+
+fn test_main() {
+	vec := Vec3([1, 2, 3]!)
+	a := vec.add(vec)
+	b := vec + vec
+	assert a == b
+}

